kube-core/src/params.rs

```diff
@@ -403,7 +447,8 @@ pub struct Preconditions {
     pub uid: Option<String>,
 }
 
-/// Propagation policy when deleting single objects
+/// Propagation policy when deleting single objects.
+/// The default used by the Kubernetes API Server varies based on resource.
```

Comment:
Nit: Maybe the note about the default should go on `DeleteParams::propagation_policy` instead, since that is where it actually becomes optional?

Reply:
Makes sense. The `DeleteParams::propagation_policy` already contains the note that "The default policy is decided by the existing finalizer set in `metadata.finalizers`, and the resource-specific default policy", which seems sufficient, so I just removed the extra comments to `PropagationPolicy`.